test/browser-headers.test.ts

```diff
@@ -215,6 +217,56 @@ describe('Browser headers', () => {
         });
     });
 
+    describe('sessionToken', () => {
+        test('gives the same headers with the same protocol', async () => {
+            generatorSpy.mockRestore();
+
+            options.resolveProtocol = () => ({ alpnProtocol: 'http/1.1' });
+
+            options.context = {
+                useHeaderGenerator: true,
+                headerGenerator,
+            };
+
+            const sessionToken = {};
+            options.context.sessionToken = sessionToken;
+
+            await browserHeadersHook(options as unknown as Options);
+            const { headers } = options;
+            options.headers = {};
+
+            await browserHeadersHook(options as unknown as Options);
+            const secondHeaders = options.headers;
+
+            expect(headers).toEqual(secondHeaders);
```

Comment:
I would add some sanity checks for the header values. This would also work if e.g. headers and secondHeaders were undefined.